tools/test-proxy/Azure.Sdk.Tools.TestProxy/Admin.cs

```diff
@@ -162,7 +162,18 @@ private object GenerateInstance(string typePrefix, string name, HashSet<string>
                 }
             }
 
-            return Activator.CreateInstance(t, arg_list.ToArray());
+            try
+            {
+                return Activator.CreateInstance(t, arg_list.ToArray());
+            }
+            catch(Exception e)
```

Comment:
When activator.CreateInstance hit's an exception during its execution, it doesn't actually throw a `HttpException` directly.  It throws an InvocationException _wrapping_ the actual exception. We handle this here before rethrowing the appropriate exception. This code means that we don't need to do any wonky untangling of exceptions on caller side.

The new section in HttpExceptionMiddleware will handle generalized exceptions.